cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/source/FileBatchSource.java

```diff
@@ -0,0 +1,210 @@
+  private static final String TABLE_DESCRIPTION = "Name of the Table that keeps track of the last time files " +
+    "were read in.";
+  private static final String INPUT_FORMAT_CLASS_DESCRIPTION = "Name of the input format class, which must be a " +
+    "subclass of FileInputFormat. Defaults to TextInputFormat";
+  private static final String FILESYSTEM_DESCRIPTION = "Distributed file system to read in from.";
+  private static final Gson GSON = new Gson();
+  private static final Logger LOG = LoggerFactory.getLogger(FileBatchSource.class);
+  private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd-HH-mm");
+
+  private static final Type MAP_STRING_STRING_TYPE = new TypeToken<Map<String, String>>() { }.getType();
+
+  private final FileBatchConfig config;
+  private KeyValueTable table;
+  private Date prevMinute;
+
+  public FileBatchSource(FileBatchConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    if (config.timeTable != null) {
+      pipelineConfigurer.createDataset(config.timeTable, KeyValueTable.class, DatasetProperties.EMPTY);
+    }
+  }
+
+  @Override
+  public void prepareRun(BatchSourceContext context) throws Exception {
+    //calculate date one minute ago, rounded down to the nearest minute
+    prevMinute = new Date(context.getLogicalStartTime() - TimeUnit.MINUTES.toMillis(1));
+    Calendar cal = Calendar.getInstance();
+    cal.setTime(prevMinute);
+    cal.set(Calendar.SECOND, 0);
+    cal.set(Calendar.MILLISECOND, 0);
+    prevMinute = cal.getTime();
+    System.out.print(FILESYSTEM_PROPERTIES_DESCRIPTION);
+    Job job = context.getHadoopJob();
+    if (config.fileSystemProperties != null) {
+      Map<String, String> properties = GSON.fromJson(config.fileSystemProperties, MAP_STRING_STRING_TYPE);
+      for (Map.Entry<String, String> entry : properties.entrySet()) {
+        job.getConfiguration().set(entry.getKey(), entry.getValue());
+      }
+    }
+
+    if (config.fileRegex != null) {
+      job.getConfiguration().set(INPUT_REGEX_CONFIG, config.fileRegex);
+    }
+    job.getConfiguration().set(INPUT_NAME_CONFIG, config.path);
+
+    if (config.timeTable != null) {
+      table = context.getDataset(config.timeTable);
+      String lastTimeRead = Bytes.toString(table.read("lastTimeRead"));
+      if (lastTimeRead == null) {
+        lastTimeRead = "0";
+      }
+      job.getConfiguration().set(LAST_TIME_READ, lastTimeRead);
+    }
+
+    job.getConfiguration().set(CUTOFF_READ_TIME, DATE_FORMAT.format(prevMinute));
+    if (config.inputFormatClass != null) {
+      ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+      Class<? extends FileInputFormat> classType = (Class<? extends FileInputFormat>)
+        classLoader.loadClass(config.inputFormatClass);
+      job.setInputFormatClass(classType);
+    }
+    FileInputFormat.setInputPathFilter(job, BatchFileFilter.class);
```

Comment:
Eventually, I think we need to allow users to specify their own Filters. Should probably add a jira to allow this to be configured as well, in a future PR. What do you think?